### ISSUE TITLE
 app crash when mode terminate

### ISSUE THREAD

**Issue by mudiono1413** [transistorsoft/flutter_background_geolocation#182]
i was following doc Android Headless Mode ,
but doesnt work, my apps crash when terminate

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
    runApp(new MaterialApp(
      title: 'My Aps',
      debugShowCheckedModeBanner: false,
      theme: new ThemeData(
        primarySwatch: Colors.indigo,
        primaryColor: const Color(0xFF4776CD),
        primaryColorLight: Colors.white,
        primaryColorDark: Colors.black,
        fontFamily: "Poppins",
        hintColor: Colors.black,
        inputDecorationTheme: new InputDecorationTheme(
          labelStyle: new TextStyle(
              color: Colors.white, fontFamily: 'Poppins', fontSize: 20.0),
          hintStyle: new TextStyle(
              fontSize: 16.0, color: Colors.grey[300], fontFamily: 'Poppins'),
        ),
      ),
      home: new SplashScreen(),
  
    ));
    bg.BackgroundGeolocation.registerHeadlessTask(headlessTask);
}

**Comment by christocracy** [comment]
*****************************************************************
*
* WARNING:  DO NOT IGNORE THIS TEMPLATE
*
*****************************************************************



## Your Environment
* Plugin version:
* Platform: iOS or Android
* OS version:
* Device manufacturer / model:
* Flutter info (`flutter doctor`):
* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
PASTE_YOUR_CODE_HERE
```

## Expected Behavior


## Actual Behavior


## Steps to Reproduce

1.
2.
3.
4.

## Context


## Debug logs

<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
PASTE_YOUR_LOGS_HERE
```

</details>

**Comment by christocracy** [comment]
Silently closed without explanation?

**Comment by vijayAndroidDev** [comment]
Hi, am using background location in the flutter mobile app. when the app goes to the background Headless State begins, after the activity change for example still to moving state. I got this error. @christocracy 

 D/TSLocationManager: [c.t.l.service.AbstractService a] 
      🎾  HeartbeatService [eventCount: 1]
 D/TSLocationManager: [c.t.l.s.HeartbeatService$a run] ❤️
 W/MediaPlayerNative: info/warning (10973, 0)
 D/TSLocationManager: [c.t.l.a.BackgroundGeolocation isMainActivityActive] NO
 D/TSLocationManager: [c.t.f.b.HeadlessTask onHeadlessEvent] 💀 [HeadlessTask heartbeat]
 D/TSLocationManager: [c.t.l.service.AbstractService a] ⚙️︎  finish HeartbeatService [eventCount: 0, sticky: false]
 D/TSLocationManager: [c.t.f.b.HeadlessTask dispatch] [HeadlessTask] waiting for client to initialize
 D/TSLocationManager: [c.t.l.service.AbstractService onDestroy] 
      🔴  HeartbeatService destroyed

**Comment by christocracy** [comment]
There is no error in those logs.

that is all normal and expected.

yes, “service destroyed” is normal and red simply means OFF.

the services are not required to stay alive once their work is complete.

**Comment by vijayAndroidDev** [comment]
Ok. But the Requirement is to update the current location to my server even app is in a kill state. how to implement this requirement. now am trying Heartbeat callback on foreground and background. when the heartbeat is destroyed in the background how to implement this?? plz give the solution. @christocracy

**Comment by christocracy** [comment]
If you have a specific question unrelated to this closed issue, post a new issue for yourself (do *not* ignore the issue template).
